### PR TITLE
feat: use placeholder instead image

### PR DESCRIPTION
**Pull request by ferrarafer**
- Replace real images with local placeholders
- Stall the app until session selected
- Improved data masking

NOTE: When I was wrapping the changes, I exceeded book api quota 😄 , I will double check all tomorrow, maybe will be better to avoid merge till tomorrow.

> Exception: Quota exceeded for quota metric 'Queries' and limit 'Queries per day' of service 'books.googleapis.com' for consumer.

**DEMO**

https://github.com/FilledStacks/session_mate/assets/1505545/78886fd0-2083-4b7b-ba36-acf7e4531d35

https://github.com/FilledStacks/session_mate/assets/1505545/988934c8-cbd0-4b8f-897c-d0f046cbb516

### CODE REVIEW

**Review by FilledStacks — CHANGES_REQUESTED** [review]
This looks really great! I'm happy with it. 

One thing we need to do is to stall the app until we tap the play button, not until we select the session. 

Reason: we only need to fill the cache at that point since we might be tapping through different sessions before we select one to play.

**Comment by ferrarafer** [comment]
**New Demo**

https://github.com/FilledStacks/session_mate/assets/1505545/9c8df236-e178-4ae3-9075-3b1f63abf72c

File: example/lib/main.dart [hunk 1]
```diff
@@ -26,11 +26,18 @@ class MainApp extends StatelessWidget {
       navigatorObservers: [
         StackedService.routeObserver,
       ],
-      builder: (context, child) => SessionMate(
+      builder: (context, child) => SessionMateBuilder(
+        excludeKeysOnDataMasking: const [
```
Comment (FilledStacks) [inline example/lib/main.dart:30]: Perfect

File: lib/src/widgets/driver_ui/driver_ui.dart [hunk 2]
```diff
@@ -9,8 +10,7 @@ import 'package:stacked/stacked.dart';
 
 class DriverUI extends StackedView<DriverUIViewModel> {
   final Widget child;
-  final VoidCallback? onRestart;
-  const DriverUI({super.key, required this.child, this.onRestart});
+  const DriverUI({super.key, required this.child});
```
Comment (FilledStacks) [inline lib/src/widgets/driver_ui/driver_ui.dart:13]: I'm very happy to see the required restart code removed

File: test/services/data_masking_service_test.dart
Comment (FilledStacks) [inline test/services/data_masking_service_test.dart:1]: 🔥

File: test/services/configuration_service_test.dart
Comment (FilledStacks) [inline test/services/configuration_service_test.dart:1]: 🔥

**Review by FilledStacks — APPROVED** [review]
Great PR! Local session replay is wrapped up nicely with these features.

**Pull request merged**